pbs/its
# Bugfix/octo 9149 better transparency detection

@M3lania pointed out that our previous logic would incorrectly refuse to convert opaque pngs to jpegs, with the format=auto parameter, if they had an alpha channel.

This patch examines the pixels of a PNG, in such a case, to check if any of the pixels are actually transparent.

# To test locally

* check this branch out: `git checkout bugfix/OCTO-9149-better-transparency-detection`
* start the application: `docker-compose up --build`
* open http://localhost:5000/tests/images/opaque_with_alpha.png?format=auto

the image should become a jpeg: this is correct behavior.

* check out former application code (to see previous behavior): `git checkout master -- its/optimize.py`
* open http://localhost:5000/tests/images/opaque_with_alpha.png?format=auto

you should see the previous behavior: the image remains a png
